internal/core/application/market_loader_service.go

```diff
@@ -62,21 +63,97 @@ func (m *marketsLoaderService) FetchMarkets() {
 		return
 	}
 
-	if err := m.marketRepository.DeleteAllMarket(context.Background()); err != nil {
-		log.Errorf("FetchMarkets -> DeleteAllMarket: %v", err)
+	//markets already stored in db
+	existingMarkets, err := m.marketRepository.GetAllMarkets(context.Background())
+	if err != nil {
+		log.Errorf("FetchMarkets -> GetAllMarkets: %v", err)
 		return
 	}
 
+	//assume that external service, from which we are fetching markets, should return active markets
+	activeMarkets := make([]domain.Market, 0)
 	for _, v := range liquidityProviders {
-		for _, v1 := range v.Markets {
-			if err := m.marketRepository.InsertMarket(context.Background(), domain.Market{
+		for _, market := range v.Markets {
+			activeMarkets = append(activeMarkets, domain.Market{
 				ProviderName: v.Name,
 				Url:          v.Endpoint,
-				BaseAsset:    v1.BaseAsset,
-				QuoteAsset:   v1.QuoteAsset,
-			}); err != nil {
-				log.Errorf("FetchMarkets -> InsertMarket: %v", err)
+				BaseAsset:    market.BaseAsset,
+				QuoteAsset:   market.QuoteAsset,
+				Active:       true,
+			})
+		}
+	}
+
+	if err := m.updateMarketActiveStatusAndInsertNew(activeMarkets, existingMarkets); err != nil {
+		log.Errorf("FetchMarkets -> updateMarketActiveStatusAndInsertNew: %v", err)
+	}
+}
+
+func (m *marketsLoaderService) updateMarketActiveStatusAndInsertNew(
+	existingMarkets []domain.Market,
+	activeMarkets []domain.Market,
```

Comment:
watch out! we're passing the list of active/exiting markets in the wrong order!

Reply:
Done